src/app/ecosystem-explorer/schemas/ProjectTechSchema.ts

```diff
@@ -0,0 +1,22 @@
+import { z } from 'zod'
+
+import type { CMSFieldConfig } from '@/types/cmsConfig'
+
+import configJson from '@/data/cmsConfigSchema.json'
+
+import { createCategorySchema } from '@/utils/categoryUtils'
+import { findOrThrow } from '@/utils/findOrThrow'
+
+const ecosystemCollection = findOrThrow(
+  configJson.collections,
+  (collection) => collection.name === 'ecosystem_projects',
+)
+
+const ecosystemFields = ecosystemCollection.fields as Array<CMSFieldConfig>
+
+const techField = findOrThrow(ecosystemFields, (field) => field.name === 'tech')
+const options = techField.options as unknown as Array<string>
+
+const TechOptionEnum = createCategorySchema(options)
+
+export const ProjectTechSchema = z.array(TechOptionEnum).nonempty()
```

Comment:
```suggestion
import { z } from 'zod'

import { createCategorySchema } from '@/utils/categoryUtils'
import { getCollectionConfig, getCMSFieldOptions } from '@/utils/cmsConfigUtils'

const ecosystemConfig = getCollectionConfig('ecosystem_projects')

const ecosystemFields = ecosystemConfig.fields

const techOptions = getCMSFieldOptions(
  ecosystemFields,
  'tech',
) as unknown as Array<string>

const TechOptionEnum = createCategorySchema(techOptions)

export const ProjectTechSchema = z.array(TechOptionEnum).nonempty()
```

Reply:
Isn't using `createCategorySchema` a bit confusing here? It's under `categoryUtils` - Should we abstract it into a different file - something a bit more generic? `createEnumSchema` or `createOptionsSchema`?

Reply:
Ah nice! I forgot about these CMS utility functions, I don't use them often. Thanks for the refactor, it's better like that indeed :)

And yes, let's change `createCategorySchema`. I like `createEnumSchema` because it's exactly what the function does, unlike `createCategorySchema`. Nothing in the function implementation is category-specific.

I'm gonna rename it and move it to our existing `zodUtils.ts` file